File: PersonnelManagement/PersonnelManagement/Front/classes/Subject_classes/elementsubject.ts
export default class elementsubject {
    constructor() {
        this.id = 0;
        this.subject1 = 0;
        this.subject2 = 0;
        this.subject3 = 0;
        this.subject4 = 0;
        this.subject5 = 0;
        this.subject6 = 0;
        this.subject7 = 0;
        this.subject8 = 0;
        this.subject9 = 0;
        this.subject10 = 0;
        this.subject11 = 0;
        this.subject12 = 0;
        this.subject13 = 0;
        this.subject14 = 0;
        this.subject15 = 0;
        this.subject16 = 0;
        this.subject17 = 0;
        this.subject18 = 0;
        this.subject19 = 0;
        this.subject20 = 0;
    }

    id: number;
    subject1: number;
    subject2: number;
    subject3: number;
    subject4: number;
    subject5: number;
    subject6: number;
    subject7: number;
    subject8: number;
    subject9: number;
    subject10: number;
    subject11: number;
    subject12: number;
    subject13: number;
    subject14: number;
    subject15: number;
    subject16: number;
    subject17: number;
    subject18: number;
    subject19: number;
    subject20: number;

    setSubjects(list: number[]): void {
        var len: number = list.length;
        if (len >= 1) { this.subject1 = list[0] }
        if (len >= 2) { this.subject2 = list[1] }
        if (len >= 3) { this.subject3 = list[2] }
        if (len >= 4) { this.subject4 = list[3] }
        if (len >= 5) { this.subject5 = list[4] }
        if (len >= 6) { this.subject6 = list[5] }
        if (len >= 7) { this.subject7 = list[6] }
        if (len >= 8) { this.subject8 = list[7] }
        if (len >= 9) { this.subject9 = list[8] }
        if (len >= 10) { this.subject10 = list[9] }
        if (len >= 11) { this.subject11 = list[10] }
        if (len >= 12) { this.subject12 = list[11] }
        if (len >= 13) { this.subject13 = list[12] }
        if (len >= 14) { this.subject14 = list[13] }
        if (len >= 15) { this.subject15 = list[14] }
        if (len >= 16) { this.subject16 = list[15] }
        if (len >= 17) { this.subject17 = list[16] }
        if (len >= 18) { this.subject18 = list[17] }
        if (len >= 19) { this.subject19 = list[18] }
        if (len == 20) { this.subject20 = list[19] }
    };

    getSubjectStructure(): number[] {
        var output: number[] = [
            this.subject1,
            this.subject2,
            this.subject3,
            this.subject4,
            this.subject5,
            this.subject6,
            this.subject7,
            this.subject8,
            this.subject9,
            this.subject10,
            this.subject11,
            this.subject12,
            this.subject13,
            this.subject14,
            this.subject15,
            this.subject16,
            this.subject17,
            this.subject18,
            this.subject19,
            this.subject20];
        output = output.filter(r => r > 0 );
        return output;
    };

    setElementByPromis(object) {
        this.id = object.id;
        this.subject1 = object.subject1;
        this.subject2 = object.subject2;
        this.subject3 = object.subject3;
        this.subject4 = object.subject4;
        this.subject5 = object.subject5;
        this.subject6 = object.subject6;
        this.subject7 = object.subject7;
        this.subject8 = object.subject8;
        this.subject9 = object.subject9;
        this.subject10 = object.subject10;
        this.subject11 = object.subject11;
        this.subject12 = object.subject12;
        this.subject13 = object.subject13;
        this.subject14 = object.subject14;
        this.subject15 = object.subject15;
        this.subject16 = object.subject16;
        this.subject17 = object.subject17;
        this.subject18 = object.subject18;
        this.subject19 = object.subject19;
        this.subject20 = object.subject20;
    };
}

File: PersonnelManagement/PersonnelManagement/Models/Subjects/Elementsubject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace PersonnelManagement.Models
{
    public partial class Elementsubject
    {
        public Elementsubject()
        {
            Subject1 = 0;
            Subject2 = 0;
            Subject3 = 0;
            Subject4 = 0;
            Subject5 = 0;
            Subject6 = 0;
            Subject7 = 0;
            Subject8 = 0;
            Subject9 = 0;
            Subject10 = 0;
            Subject11 = 0;
            Subject12 = 0;
            Subject13 = 0;
            Subject14 = 0;
            Subject15 = 0;
            Subject16 = 0;
            Subject17 = 0;
            Subject18 = 0;
            Subject19 = 0;
            Subject20 = 0;
        }
        public Elementsubject(Elementsubject elementsubject)
        {
            Id = elementsubject.Id;
            Subject1 = elementsubject.Subject1;
            Subject2 = elementsubject.Subject2;
            Subject3 = elementsubject.Subject3;
            Subject4 = elementsubject.Subject4;
            Subject5 = elementsubject.Subject5;
            Subject6 = elementsubject.Subject6;
            Subject7 = elementsubject.Subject7;
            Subject8 = elementsubject.Subject8;
            Subject9 = elementsubject.Subject9;
            Subject10 = elementsubject.Subject10;
            Subject11 = elementsubject.Subject11;
            Subject12 = elementsubject.Subject12;
            Subject13 = elementsubject.Subject13;
            Subject14 = elementsubject.Subject14;
            Subject15 = elementsubject.Subject15;
            Subject16 = elementsubject.Subject16;
            Subject17 = elementsubject.Subject17;
            Subject18 = elementsubject.Subject18;
            Subject19 = elementsubject.Subject19;
            Subject20 = elementsubject.Subject20;
        }
        public Elementsubject(Structure structure)
        {
            Subject1 = (uint)structure.Subject1;
            Subject2 = (uint)structure.Subject2;
            Subject3 = (uint)structure.Subject3;
            Subject4 = (uint)structure.Subject4;
            Subject5 = (uint)structure.Subject5;
            Subject6 = (uint)structure.Subject6;
            Subject7 = (uint)structure.Subject7;
            Subject8 = (uint)structure.Subject8;
            Subject9 = (uint)structure.Subject9;
            Subject10 = (uint)structure.Subject10;
            Subject11 = (uint)structure.Subject11;
            Subject12 = (uint)structure.Subject12;
            Subject13 = (uint)structure.Subject13;
            Subject14 = (uint)structure.Subject14;
            Subject15 = (uint)structure.Subject15;
        }

        public Elementsubject(Position position)
        {
            Subject1 = (uint)position.Subject1;
            Subject2 = (uint)position.Subject2;
            Subject3 = (uint)position.Subject3;
            Subject4 = (uint)position.Subject4;
            Subject5 = (uint)position.Subject5;
            Subject6 = (uint)position.Subject6;
            Subject7 = (uint)position.Subject7;
            Subject8 = (uint)position.Subject8;
            Subject9 = (uint)position.Subject9;
            Subject10 = (uint)position.Subject10;
            Subject11 = (uint)position.Subject11;
            Subject12 = (uint)position.Subject12;
            Subject13 = (uint)position.Subject13;
            Subject14 = (uint)position.Subject14;
            Subject15 = (uint)position.Subject15;
            Subject16 = (uint)position.Subject16;
            Subject17 = (uint)position.Subject17;
            Subject18 = (uint)position.Subject18;
            Subject19 = (uint)position.Subject19;
            Subject20 = (uint)position.Subject20;
        }

        /*public string getFullName*/
    }
}
